cannon/mipsevm/tests/evm_multithreaded_test.go

```diff
@@ -249,70 +249,64 @@ func TestEVM_SysClone_FlagHandling(t *testing.T) {
+		mtutil.InitializeSingleThread(r.Intn(10000), state, c.traverseRight)
+		testutil.StoreInstruction(state.Memory, state.GetPC(), syscallInsn)
+		state.GetRegistersRef()[2] = arch.SysClone        // the syscall number
+		state.GetRegistersRef()[4] = exec.ValidCloneFlags // a0 - first argument, clone flags
+		state.GetRegistersRef()[5] = stackPtr             // a1 - the stack pointer
 
-				// Sanity-check assumptions
-				require.Equal(t, Word(1), state.NextThreadId)
+		// Sanity-check assumptions
+		require.Equal(t, Word(1), state.NextThreadId)
+	}
 
-				// Setup expectations
-				expected := mtutil.NewExpectedState(t, state)
-				expected.Step += 1
-				expectedNewThread := expected.ExpectNewThread()
-				expected.ActiveThreadId = expectedNewThread.ThreadId
-				expected.StepsSinceLastContextSwitch = 0
-				if c.traverseRight {
-					expected.RightStackSize += 1
-				} else {
-					expected.LeftStackSize += 1
-				}
-				// Original thread expectations
-				expected.PrestateActiveThread().PC = state.GetCpu().NextPC
-				expected.PrestateActiveThread().NextPC = state.GetCpu().NextPC + 4
-				expected.PrestateActiveThread().Registers[2] = 1
-				expected.PrestateActiveThread().Registers[7] = 0
-				// New thread expectations
-				expectedNewThread.PC = state.GetCpu().NextPC
-				expectedNewThread.NextPC = state.GetCpu().NextPC + 4
-				expectedNewThread.ThreadId = 1
-				expectedNewThread.Registers[register.RegSyscallRet1] = 0
-				expectedNewThread.Registers[register.RegSyscallErrno] = 0
-				expectedNewThread.Registers[register.RegSP] = stackPtr
+	setExpectations := func(c testCase, expected *mtutil.ExpectedState, vm VersionedVMTestCase) ExpectedExecResult {
+		expected.Step += 1
+		expectedNewThread := expected.ExpectNewThread()
+		expected.ActiveThreadId = expectedNewThread.ThreadId
+		expected.StepsSinceLastContextSwitch = 0
+		if c.traverseRight {
+			expected.RightStackSize += 1
+		} else {
+			expected.LeftStackSize += 1
+		}
 
-				var err error
-				var stepWitness *mipsevm.StepWitness
-				stepWitness, err = goVm.Step(true)
-				require.NoError(t, err)
+		// Original thread expectations
+		prestateNextPC := expected.ActiveThread().NextPC
+		expected.PrestateActiveThread().PC = prestateNextPC
+		expected.PrestateActiveThread().NextPC = prestateNextPC + 4
+		expected.PrestateActiveThread().Registers[2] = 1
+		expected.PrestateActiveThread().Registers[7] = 0
+		// New thread expectations
+		expectedNewThread.PC = prestateNextPC
+		expectedNewThread.NextPC = prestateNextPC + 4
+		expectedNewThread.ThreadId = 1
+		expectedNewThread.Registers[register.RegSyscallRet1] = 0
+		expectedNewThread.Registers[register.RegSyscallErrno] = 0
+		expectedNewThread.Registers[register.RegSP] = stackPtr
 
-				expected.Validate(t, state)
-				activeStack, inactiveStack := mtutil.GetThreadStacks(state)
-				require.Equal(t, 2, len(activeStack))
-				require.Equal(t, 0, len(inactiveStack))
```

Comment:
I originally added a `PostCheck` to validate these lines, but looking again these assertions should already be covered by `expected.ExpectNewThread()`.  To keep the API simpler, I've cut these redundant checks.